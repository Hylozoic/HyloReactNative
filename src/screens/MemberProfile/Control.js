import React, { useRef } from 'react'
import {
  View,
  TouchableOpacity,
  TextInput,
  Text
} from 'react-native'
import EntypoIcon from 'react-native-vector-icons/Entypo'
import { rhino60, amaranth, rhino50 } from 'style/colors'

export default function Control ({
  value,
  onChangeText,
  editable = false,
  multiline,
  isMe,
  onPress,
  hideEditIcon,
  error,
  placeholder,
  style
}) {
  const inputRef = useRef()
  const focus = () => inputRef.current?.focus()

  return (
    <View style={[styles.control, editable && styles.editableControl]}>
      <View style={styles.controlInputRow}>
        {!editable && !!value && (
          <Text style={style}>{value}</Text>
        )}
        {editable && (<>
          <TextInput
            ref={inputRef}
            onFocus={onPress}
            style={[styles.controlInput, style]}
            value={value}
            onChangeText={onChangeText}
            editable={editable}
            placeholder={isMe ? placeholder : ''}
            multiline={multiline}
            numberOfLines={multiline ? 8 : 1}
            underlineColorAndroid='transparent'
          />
          {!hideEditIcon && (
            <TouchableOpacity onPress={focus} style={styles.editIconWrapper}>
              <EntypoIcon name='edit' style={styles.editIcon} />
            </TouchableOpacity>
          )}  
        </>)}
      </View>
      {!!error && (
        <View style={styles.controlError}>
          <Text style={styles.controlErrorText}>{error}</Text>
        </View>
      )}
    </View>
  )
}

const styles = {
  control: {
    flex: 1
  },
  editableControl: {
    marginBottom: 5
  },
  controlInputRow: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  controlInput: {
    textAlignVertical: 'top',
    flex: 1
  },
  editIconWrapper: {
    marginRight: 10
  },
  editIcon: {
    color: rhino60,
    fontSize: 16
  },
  controlError: {
    paddingTop: 5,
    paddingBottom: 3
  },
  controlErrorText: {
    fontSize: 13,
    fontFamily: 'Circular-Book',
    color: amaranth
  }
}
