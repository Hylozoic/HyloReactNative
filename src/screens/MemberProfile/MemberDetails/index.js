// import { withNavigationFocus } from '@react-navigation/compat'
// import component from './MemberDetails'
// import connector from './MemberDetails.connector'

// export default withNavigationFocus(connector(component))

import component from './MemberDetails'

export default component
