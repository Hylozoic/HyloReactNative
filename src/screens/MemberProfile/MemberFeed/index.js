import component from './MemberFeed'
import connector from './MemberFeed.connector'

export default connector(component)
