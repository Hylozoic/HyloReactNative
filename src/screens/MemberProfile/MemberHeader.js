import React from 'react'
import {
  View,
  TouchableOpacity,
  TextInput,
  Text,
  Alert
} from 'react-native'
import Icon from 'components/Icon'
import EntypoIcon from 'react-native-vector-icons/Entypo'
import PopupMenuButton from 'components/PopupMenuButton'
import { filter, get, isEmpty } from 'lodash/fp'
import styles from './MemberHeader.styles'
import { AXOLOTL_ID } from 'store/models/Person'

export default function MemberHeader ({
  person,
  flagMember,
  onPressMessages,
  isMe,
  editProfile,
  editAccount,
  editable,
  updateSetting = () => {},
  saveChanges,
  errors = {},
  ...props
}) {
  if (!person) return null

  const { name, location, tagline } = person
  const blockUser = blockUserWithConfirmationFun(props.blockUser, name)
  const isAxolotl = AXOLOTL_ID === get('id', person)

  return (
    <View style={styles.header}>
      <View style={styles.nameRow}>
        <Control
          style={styles.name}
          value={name}
          placeholder='Name'
          editable={editable}
          onChangeText={updateSetting('name')}
          error={errors.name}
          isMe={isMe}
        />
        <View style={styles.icons}>
          {!isMe &&<TouchableOpacity onPress={onPressMessages}>
            <Icon name='Messages' style={styles.icon} />
          </TouchableOpacity>}
          <MemberMenu {...{
            flagMember, isMe, editProfile, editAccount,
            saveChanges, editable, blockUser, isAxolotl
          }} />
        </View>
      </View>
      <Control
        style={styles.location}
        value={location}
        placeholder='Location'
        editable={editable}
        onChangeText={updateSetting('location')}
        isMe={isMe}
      />
      <Control
        style={styles.tagline}
        value={tagline}
        placeholder='Short Description'
        editable={editable}
        onChangeText={updateSetting('tagline')}
        isMe={isMe}
      />
    </View>
  )
}

export function blockUserWithConfirmationFun (blockUserFun, name) {
  return function () {
    return Alert.alert(
      `Are you sure you want to block ${name}?`,
      `You will no longer see ${name}\'s activity
      and they won't see yours.
      
      You can unblock this member at any time.
      Go to Settings > Blocked Users.`,
      [
        { text: `Block ${name}`, onPress: (blockedUserId) => blockUserFun(blockedUserId) },
        { text: 'Cancel', style: 'cancel' }
      ])
  }
}
export class Control extends React.Component {
  inputRef = React.createRef()

  focus = () => this.inputRef.current && this.inputRef.current.focus()

  render () {
    const {
      value, onChangeText, editable = false, style, multiline,
      hideEditIcon, error, placeholder, isMe
    } = this.props
    return (
      <View style={[styles.control, editable && styles.editableControl]}>
        <View style={styles.controlInputRow}>
          {editable || !multiline
            ? <TextInput
                ref={this.inputRef}
                style={[styles.controlInput, style]}
                value={value}
                onChangeText={onChangeText}
                editable={editable}
                placeholder={isMe ? placeholder : ''}
                multiline={multiline}
                numberOfLines={multiline ? 8 : 1}
                underlineColorAndroid='transparent'
              />
            : <Text>{value}</Text>}
          {editable && !hideEditIcon && <TouchableOpacity onPress={this.focus} style={styles.editIconWrapper}>
            <EntypoIcon name='edit' style={styles.editIcon} />
          </TouchableOpacity>}
        </View>
        {!!error && <View style={styles.controlError}><Text style={styles.controlErrorText}>{error}</Text></View>}
      </View>
    )
  }
}

export function MemberMenu ({
  flagMember, isMe, blockUser, editProfile,
  editAccount, saveChanges, editable, isAxolotl
}) {
  // If the function is defined, than it's a valid action

  const actions = filter(x => x[1], [
    ['Edit Profile', isMe && !editable && editProfile],
    ['Edit Account', isMe && !editable && editAccount],
    ['Save Changes', isMe && editable && saveChanges],
    ['Flag This Member', !isMe && flagMember],
    ['Block This Member', !isMe && !isAxolotl && blockUser]
  ])

  if (isEmpty(actions)) return null

  const destructiveButtonIndex = get('1.0', actions) === 'Block This Member' ? 1 : -1

  return (
    <PopupMenuButton
      actions={actions}
      destructiveButtonIndex={destructiveButtonIndex}
    >
      <Icon name='More' style={styles.lastIcon} />
    </PopupMenuButton>
  )
}
