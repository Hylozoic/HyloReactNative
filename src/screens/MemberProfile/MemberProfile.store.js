import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'

export const MODULE_NAME = 'MemberProfile'
export const FETCH_PERSON = `${MODULE_NAME}/FETCH_PERSON`

export function fetchPerson (id) {
  return {
    type: FETCH_PERSON,
    graphql: {
      query: `query PersonDetails ($id: ID) {
        person (id: $id) {
          id
          name
          avatarUrl
          bannerUrl
          bio
          twitterName
          linkedinUrl
          facebookUrl
          url
          tagline
          location
          messageThreadId
          memberships {
            id
            role
            hasModeratorRole
            community {
              id
              name
              slug
              memberCount
            }
          }
          skills (first: 100) {
            total
            hasMore
            items {
              id
              name
            }
          }
        }
      }`,
      variables: { id }
    },
    meta: {
      afterInteractions: true,
      extractModel: 'Person'
    }
  }
}

export const getPerson = ormCreateSelector(
  orm,
  (_, { id }) => id,
  (session, id) => {
    const person = session.Person.safeGet({ id })
    if (!person) return null
    return {
      ...person.ref,
      skills: person.skills.toRefArray(),
      memberships: person.memberships.toModelArray()
    }
  }
)
