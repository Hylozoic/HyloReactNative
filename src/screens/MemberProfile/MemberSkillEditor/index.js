import React from 'react'
import { ScrollView } from 'react-native'
import KeyboardFriendlyView from 'components/KeyboardFriendlyView'
import SkillEditor from 'components/SkillEditor'
import styles from './MemberSkillEditor.styles'

export default function MemberSkillEditor ({ navigation }) {    
  return (
    <KeyboardFriendlyView style={styles.container}>
      <ScrollView>
        <SkillEditor done={() => navigation.goBack()} doneLabel='Done' theme={styles.skillEditor} />
      </ScrollView>
    </KeyboardFriendlyView>
  )
}
