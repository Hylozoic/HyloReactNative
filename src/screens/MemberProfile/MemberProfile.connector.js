import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { get } from 'lodash/fp'
import { fetchPerson } from './MemberProfile.store'
import blockUser from 'store/actions/blockUser'
import makeGoToGroup from 'store/actions/makeGoToGroup'
import getMe from 'store/selectors/getMe'
import getPerson from 'store/selectors/getPerson'
import getBlockedUsers from 'store/selectors/getBlockedUsers'
import updateUserSettings from 'store/actions/updateUserSettings'
import { mapWhenFocused, mergeWhenFocused } from 'util/redux'
import getCurrentGroup from 'store/selectors/getCurrentGroup'
import { logout } from 'screens/Login/actions'

export function mapStateToProps (state, props) {
  const currentUser = getMe(state, props)
  const currentGroup = getCurrentGroup(state, props)
  const id = get('route.params.id', props)
  const person = id
    ? getPerson(state, { personId: id })
    : currentUser
  const isMe = Number(get('id', currentUser)) === Number(get('id', person))

  const editing = get('route.params.editing', props)
  const isBlocked = !!getBlockedUsers(state).find(i => get('id', i) === id)
  const goToDetails = () => props.navigation.navigate('MemberDetails', { id })
  const goToEdit = () => props.navigation.navigate('MemberDetails', { id, editing: true })
  const goToEditAccount = () => props.navigation.navigate('Edit Account Info')
  const goToSkills = () => props.navigation.navigate('MemberSkillEditor', { id })
  const navigation = props.navigation

  return {
    isBlocked,
    id,
    editing,
    person,
    currentUser,
    currentGroup,
    goToDetails,
    goToEdit,
    goToEditAccount,
    goToSkills,
    isMe,
    navigation
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToGroup: makeGoToGroup(),
    ...bindActionCreators({
      fetchPerson,
      updateUserSettings,
      blockUser,
      logout
    }, dispatch)
  }
}

export function makeOnPressMessages (currentUser, person, navigation) {
  if (!person || currentUser.id === person.id) return () => navigation.navigate('Messages')
  const { messageThreadId } = person
  if (messageThreadId) return () => navigation.navigate('Messages', { screen: 'Thread', params: { id: messageThreadId }})
  return () => navigation.navigate('Messages', { screen: 'New Message', params: { participantIds: [person.id] } })
}

export function mergeProps (stateProps, dispatchProps, ownProps) {
  const { id, currentUser, person, isMe } = stateProps
  const { navigation } = ownProps

  const fetchPerson = () => dispatchProps.fetchPerson(id)

  const canFlag = currentUser && id && currentUser.id !== id

  const onPressMessages = makeOnPressMessages(currentUser, person, navigation)

  // only call updateUserSettings if isMe (otherwise when you visit another
  // user and then press the back button, you will update your profile with their
  // settings)
  const updateUserSettings = isMe ? dispatchProps.updateUserSettings : () => {}

  const goToMemberProfile = () => navigation.navigate('Member', { id })

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    updateUserSettings,
    canFlag,
    fetchPerson,
    onPressMessages,
    goToMemberProfile
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
