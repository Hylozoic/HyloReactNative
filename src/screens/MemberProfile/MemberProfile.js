import React from 'react'
import { Text, View, Image, TouchableOpacity } from 'react-native'
import Loading from 'components/Loading'
import styles from './MemberProfile.styles'
import MemberFeed from './MemberFeed'
import MemberHeader from './MemberHeader'
import ImagePicker from 'components/ImagePicker'
import FlagContent from 'components/FlagContent'
import EntypoIcon from 'react-native-vector-icons/Entypo'
import defaultBanner from 'assets/default-user-banner.jpg'

export default class MemberProfile extends React.Component {
  state = {
    flaggingVisible: false
  }

  componentDidMount () {
    if (this.props.isBlocked) return this.props.navigation.goBack()
    this.props.fetchPerson()
    this.setHeader()
  }

  setHeader = () => {
    const { navigation, currentGroup } = this.props
    navigation.setOptions({ title: currentGroup.name  })
  }

  blockUser = () => {
    const { person, blockUser } = this.props
    blockUser(person.id)
      .then(() => this.props.navigation.goBack())
  }

  render () {
    const {
      isBlocked,
      canFlag,
      goToDetails,
      goToEdit,
      goToEditAccount,
      goToManageNotifications,
      goToBlockedUsers,
      id,
      isFocused,
      isMe,
      onPressMessages,
      person,
      updateUserSettings,
      navigation
    } = this.props
    const { flaggingVisible } = this.state

    if (!person) return <Loading />

    let flagMember
    if (canFlag) {
      flagMember = () => {
        this.setState({ flaggingVisible: true })
      }
    }

    // Used to generate a link to this post from the backend.
    const linkData = {
      id,
      type: 'member'
    }
    const header = (
      <View>
        <MemberBanner
          isFocused={isFocused}
          isMe={isMe}
          person={person}
          updateUserSettings={updateUserSettings}
        />
        <View style={styles.marginContainer}>
          <MemberHeader
            person={person}
            flagMember={flagMember}
            blockUser={this.blockUser}
            onPressMessages={onPressMessages}
            isMe={isMe}
            goToEdit={goToEdit}
            goToEditAccount={goToEditAccount}
            goToManageNotifications={goToManageNotifications}
            goToBlockedUsers={goToBlockedUsers}
          />
          <ReadMoreButton goToDetails={goToDetails} />
        </View>
        {flaggingVisible && (
          <FlagContent type='member' linkData={linkData}
            onClose={() => this.setState({ flaggingVisible: false })} />
        )}
      </View>
    )

    return <MemberFeed id={id} header={header} navigation={navigation} />
  }
}

export class MemberBanner extends React.Component {
  state = {
    avatarPickerPending: false,
    bannerPickerPending: false,
    avatarLocalUri: null,
    bannerLocalUri: null
  }

  onChoice ({ local, remote }, prefix) {
    const localKey = `${prefix}LocalUri`
    const remoteKey = `${prefix}Url`
    this.setState({
      [localKey]: local
    })

    this.props.updateUserSettings({ [remoteKey]: remote })
  }

  render () {
    const { person: { id, avatarUrl, bannerUrl }, isMe } = this.props
    const { avatarPickerPending, bannerPickerPending, avatarLocalUri, bannerLocalUri } = this.state

    const avatarSource = avatarLocalUri
      ? { uri: avatarLocalUri }
      : avatarUrl && { uri: avatarUrl }

    // This is a suprisingly annoying piece of logic. Basically, prefer
    // displaying `bannerLocalUri`, then `bannerUrl`, then `defaultBanner`.
    // However, don't display `defaultBanner` only to be replaced with
    // `bannerUrl` after the request finishes! The trick to it is this:
    // `bannerUrl` will be undefined until the request finishes, then it should
    // be either string or null. So we don't display the default unless it's null.

    let bannerSource = {
      uri: bannerLocalUri || bannerUrl
    }
    if (bannerUrl === null && bannerLocalUri === null) {
      bannerSource = defaultBanner
    }

    return (
      <View>
        <ImagePicker
          title='Change Banner'
          type='userBanner'
          style={styles.bannerImagePicker}
          id={id}
          onChoice={choice => this.onChoice(choice, 'banner')}
          onPendingChange={pending => this.setState({ bannerPickerPending: pending })}
          disabled={!isMe}
        >
          <Image source={bannerSource} style={styles.bannerImage} />
          {isMe && <EditButton isLoading={bannerPickerPending} style={styles.bannerEditButton} />}
        </ImagePicker>
        <View style={styles.avatarW3}>
          <ImagePicker
            style={styles.avatarWrapperWrapper}
            title='Change Avatar'
            type='userAvatar'
            id={id}
            onChoice={choice => this.onChoice(choice, 'avatar')}
            onPendingChange={pending => this.setState({ avatarPickerPending: pending })}
            disabled={!isMe}
          >
            <View style={styles.avatarWrapper}>
              <Image source={avatarSource} style={styles.avatarImage} />
              {isMe && <EditButton isLoading={avatarPickerPending} style={styles.avatarEditButton} />}
            </View>
          </ImagePicker>
        </View>
      </View>
    )
  }
}

export function EditButton ({ isLoading, style }) {
  return (
    <View style={[styles.editButton, style]}>
      {isLoading
        ? <Text style={styles.editButtonText}>loading</Text>
        : <View style={{ flexDirection: 'row' }}>
          <EntypoIcon name='edit' style={styles.editIcon} />
          <Text style={styles.editButtonText}>edit</Text>
        </View>}
    </View>
  )
}

export function ReadMoreButton ({ goToDetails }) {
  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity onPress={goToDetails} style={styles.buttonWrapper}>
        <View style={styles.button}>
          <Text style={styles.buttonText}>Read More</Text>
        </View>
      </TouchableOpacity>
    </View>
  )
}
