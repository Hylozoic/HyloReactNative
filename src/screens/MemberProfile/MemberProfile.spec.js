import React from 'react'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react-native'
import { createMockStore } from 'util/testing'
import getEmptyState from 'store/getEmptyState'
import MemberProfile from 'screens/MemberProfile'

jest.mock('hooks/useGroupSelect')

describe('MemberProfile Specification', () => {
  afterEach(cleanup)

  it('default render matches snapshot', async () => {
    const navigation = {
      navigate: jest.fn(),
      setOptions: jest.fn()
    }
    const state = getEmptyState()
    const { toJSON } = render(
      <Provider store={createMockStore(state)}>
        <MemberProfile navigation={navigation} route={{ name: 'test' }} />
      </Provider>
    )

    expect(await toJSON()).toMatchSnapshot()
  })
})
