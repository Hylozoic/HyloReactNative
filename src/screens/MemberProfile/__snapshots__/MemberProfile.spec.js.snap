// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <ActivityIndicator
              color="#818A88"
              size="small"
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
          >
            <View
              collapsable={false}
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                >
                  Public Groups & Posts
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Public Groups & Posts, tab, 1 of 1"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "rgb(0, 122, 255)",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#8E8E8F",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "rgb(0, 122, 255)",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Public Groups & Posts
        </Text>
      </View>
    </View>
  </View>
</View>
`;
